sample-code/java/src/AndroidBasicInteractionsTest.java

```diff
@@ -0,0 +1,74 @@
+import io.appium.java_client.android.Activity;
+import io.appium.java_client.android.AndroidDriver;
+import io.appium.java_client.android.AndroidElement;
+import io.appium.java_client.service.local.AppiumDriverLocalService;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+public class AndroidBasicInteractionsTest {
+    private AndroidDriver<WebElement> driver;
+    private static AppiumDriverLocalService service;
+    private final String SEARCH_ACTIVITY = ".app.SearchInvoke";
+    private final String ALERT_DIALOG_ACTIVITY = ".app.AlertDialogSamples";
+    private final String PACKAGE = "io.appium.android.apis";
+
+    @BeforeSuite
+    public void setUp() throws Exception {
+        File classpathRoot = new File(System.getProperty("user.dir"));
+        File appDir = new File(classpathRoot, "../apps");
+        File app = new File(appDir.getCanonicalPath(), "ApiDemos-debug.apk");
+        service = AppiumDriverLocalService.buildDefaultService();
+        service.start();
+        DesiredCapabilities capabilities = new DesiredCapabilities();
+        capabilities.setCapability("deviceName", "Android Emulator");
+        capabilities.setCapability("app", app.getAbsolutePath());
+        capabilities.setCapability("appPackage", PACKAGE);
+        capabilities.setCapability("appActivity", SEARCH_ACTIVITY);
+        driver = new AndroidDriver<WebElement>(service.getUrl(), capabilities);
+    }
+
+    @AfterSuite
+    public void tearDown() {
+        driver.quit();
+    }
```

Comment:
Ideally we should stop the service too. @dpgraham Have another query, are we running these tests in Travis? Or it's just available as sample for reference?

Reply:
This is just for sample but if we could run it in Travis that would be a bonus.

Reply:
@dpgraham we can do that too.